src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -829,6 +829,32 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
+        public static void Regression38565_Serialize()
+        {
+            var value = new Regression38565_Parent()
+            {
+                Test = "value1",
+                Child = new Regression38565_Child()
+            };
+
+            var actual = JsonSerializer.ToString(value, new JsonSerializerOptions { IgnoreNullValues = true });
+            Assert.Equal("{\"Test\":\"value1\",\"Dict\":null,\"Child\":{\"Dict\":null}}", actual);
```

Comment:
Currently the serializer emits `"Name":null` for dictionaries which are `null`. Given that we have `IgnoreNullValues = true`, I imagine this shouldn't happen.

CC. @steveharter

Reply:
Also, CC. @layomia

Reply:
Dictionaries should support `IgnoreNullValues` just like other collection types.

Reply:
Updated the PR to respect `IgnoreNullValues` for dictionaries.

Reply:
I was just discussing this with @pgovind - do we know if `IgnoreNullValues` is being honored for other types as well (not just collections and primitives)? We should add tests for these.